docs/src/main/asciidoc/hibernate-orm.adoc

```diff
@@ -542,3 +542,219 @@ Datasource configuration is extremely simple, but is covered in a different guid
+INSERT INTO base.known_fruits(id, name) VALUES (3, 'Banana');
+
+CREATE SEQUENCE mycompany.known_fruits_id_seq;
+SELECT setval('mycompany."known_fruits_id_seq"', 3);
+CREATE TABLE mycompany.known_fruits
+(
+  id   INT,
+  name VARCHAR(40)
+);
+INSERT INTO mycompany.known_fruits(id, name) VALUES (1, 'Avocado');
+INSERT INTO mycompany.known_fruits(id, name) VALUES (2, 'Apricots');
+INSERT INTO mycompany.known_fruits(id, name) VALUES (3, 'Blackberries');
+----
+
+
+
+==== DATABASE approach
+
+For every tenant you need to create a named data source with the same identifier that is returned by the `TenantResolver`. 
+
+[source,properties]
+----
+# Disable generation
+quarkus.hibernate-orm.database.generation=none
+
+# Enable DATABASE approach
+quarkus.hibernate-orm.multitenant=DATABASE
+
+# Default tenant 'base'
+quarkus.datasource.base.db-kind=postgresql
+quarkus.datasource.base.username=quarkus_test
+quarkus.datasource.base.password=quarkus_test
+quarkus.datasource.base.jdbc.url=jdbc:postgresql://localhost:5432/quarkus_test
+
+# Tenant 'mycompany'
+quarkus.datasource.mycompany.db-kind=postgresql
+quarkus.datasource.mycompany.username=mycompany
+quarkus.datasource.mycompany.password=mycompany
+quarkus.datasource.mycompany.jdbc.url=jdbc:postgresql://localhost:5433/mycompany
+
+# Flyway configuration for the default datasource
+quarkus.flyway.locations=classpath:database/default
+quarkus.flyway.migrate-at-start=true
+
+# Flyway configuration for the mycompany datasource
+quarkus.flyway.mycompany.locations=classpath:database/mycompany
+quarkus.flyway.mycompany.migrate-at-start=true
+----
+
+Following are examples of the Flyway SQL files to be created in the configured folder `src/main/resources/database`.
+
+Default schema (`src/main/resources/database/default/V1.0.0__create_fruits.sql`):
+
+[source,sql]
+----
+CREATE SEQUENCE known_fruits_id_seq;
+SELECT setval('known_fruits_id_seq', 3);
+CREATE TABLE known_fruits
+(
+  id   INT,
+  name VARCHAR(40)
+);
+INSERT INTO known_fruits(id, name) VALUES (1, 'Cherry');
+INSERT INTO known_fruits(id, name) VALUES (2, 'Apple');
+INSERT INTO known_fruits(id, name) VALUES (3, 'Banana');
+----
+
+Mycompany schema (`src/main/resources/database/mycompany/V1.0.0__create_fruits.sql`):
+
+[source,sql]
+----
+CREATE SEQUENCE known_fruits_id_seq;
+SELECT setval('known_fruits_id_seq', 3);
+CREATE TABLE known_fruits
+(
+  id   INT,
+  name VARCHAR(40)
+);
+INSERT INTO known_fruits(id, name) VALUES (1, 'Avocado');
+INSERT INTO known_fruits(id, name) VALUES (2, 'Apricots');
+INSERT INTO known_fruits(id, name) VALUES (3, 'Blackberries');
+----
+
+=== Programmatically Resolving Tenants Connections
+
+If you need a more dynamic configuration for the different tenants you want to support and don't want to end up with multiple
+entries in your configuration file, you can use the `io.quarkus.hibernate.orm.runtime.tenant.TenantConnectionResolver`.
+
+This interface allows you to dynamically create tenant connections at runtime.
```

Comment:
you're suggesting that the interface exists, but not how one creates one. Should it suggest also that we'll look for a CDI bean having such an interface?

Reply:
> you're suggesting that the interface exists, but not how one creates one. Should it suggest also that we'll look for a CDI bean having such an interface?

I enhanced the doc for the custom tenant resolver.